### ISSUE TITLE
parent could not be empty on creation

### ISSUE THREAD

**Issue by mediatainment** [benedikt/mongoid-tree#63]
Hi Benedict, 

I have mongoid v6.0.0.beta with Ruby on Rails 5.0.0 running (perfectly at the time :-) ),
but I have troubles with the 'parent_id' and it's validation:

Here is my model

```ruby
class Role
  include Mongoid::Document
  include Mongoid::Tree

  field :name, type: String

  has_and_belongs_to_many :users

  validates_presence_of :name
end


Role.create!(name: "test")

```

which causes:

```
Mongoid::Errors::Validations: 
message:
  Validation of Role failed.
summary:
  The following errors were found: Parent Parent could not be empty
resolution:
  Try persisting the document with valid data or remove the validations.


```

Any suggestions to solve that?

Thank you very much!

**Comment by mediatainment** [comment]
It seems to be, that the validation is not made by mongoid-tree.

A the only workaround I found is, to **remove all validations** in the model and reattached the one from the ::Tree module into my model:

```ruby

class Role
  include Mongoid::Document
  include Mongoid::Tree

  field :name, type: String

  has_and_belongs_to_many :users

  # remove validations to allow 'roots' in trees
  clear_validators!
  validate :position_in_tree
  
  validates_presence_of :name
end

```

**Comment by benedikt** [comment]
Hi @mediatainment,

sorry it took me a while to get back to you. This is most likely related to a [recent change](https://jira.mongodb.org/browse/MONGOID-4180) in Mongoid. The belongs_to relations now require the relation to be present. I still have to find a way to opt-out of this in Mongoid::Tree. I'll let you know as soon as I have a fix for this.

Cheers,

Benedikt

**Comment by benedikt** [comment]
Hi again @mediatainment,

it was a quick fix and I just released v2.1.0. All tests pass for mongoid 6.0.0 (master branch) right now. Please let me know if you run into any other problems with mongoid-tree on the latest version of mongoid. 

Cheers,

Benedikt
